DRILL-5130: Implement DrillValuesRel and ValuePrel as Calcite Values sub-classes

arina-ielchiieva opened the pull request:
1. Factor out common logic for DrillValuesRel and ValuesPrel into DrillValuesRelBase.
2. Revisit explainTerms to write full tuples content so consequently Values nodes with the same row type and count but different values are considered to be different.
3. Minor refactoring.

Details in Jira DRILL-5130.

arina-ielchiieva commented on exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/DrillValuesRelBase.java:
```diff
@@ -0,0 +1,44 @@
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.drill.common.JSONOptions;
+
+public abstract class DrillValuesRelBase extends AbstractRelNode {
```
Done.

jinfengni commented on exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/DrillValuesRelBase.java:
```diff
@@ -0,0 +1,46 @@
+/**
+ * Base class for logical and physical Values implemented in Drill.
+ */
+public abstract class DrillValuesRelBase extends AbstractRelNode {
```
Calcite has an abstract class Values. Neither of DrillValueRel or ValuePrel extends from that Calcite class. Will it help solve the problem if we extend from Calcite's class?

https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/rel/core/Values.java

arina-ielchiieva commented:
@jinfengni I have implemented DrillValuesRel and ValuesPrel as Calcite Values sub-classes. Also added second constructor to avoid unnecessary tuples parsing into json during copying or logical to physical values conversion. Please review the changes when possible.

arina-ielchiieva commented:
Merged into master with commit id 33682be99e719dce9cb326e2835ebc4ae434104a
